lib/nxt_http_client/client.rb

```diff
@@ -122,6 +140,26 @@ def callback_or_response(response, response_handler)
 
     def build_response_handler(handler, &block)
       response_handler = handler || dup_handler_from_class || NxtHttpClient::ResponseHandler.new
+
+      if config.json_response
+        response_handler.configure do |handler|
+          handler.on(:success) do |response|
+            response.define_singleton_method(:body) { JSON(response.response_body) }
+            response
+          end
+        end
+      end
+
+      if config.raise_response_errors
+        response_handler.configure do |handler|
+          handler.on(:error) do |response|
+            error = NxtHttpClient::Error.new(response)
+            ::Sentry.set_extras(error_details: error.to_h) if defined?(::Sentry)
```

Comment:
I'm not a fan of blindly sending request and response contents to Sentry, but this was the previous behaviour, so I've kept it. Maybe we could expose a filter proc or something later (although Sentry already has some filters built in).

Reply:
Do you think error_details is bit too generic? Should we call it `http_error_details`?


Some extra  thought: Sentry breadcrumbs can be also interesting to use.

https://docs.sentry.io/product/issues/issue-details/breadcrumbs/

Reply:
Good point, it is indeed too generic. I've found it confusing in the past. `http_error_details` is better, thanks.

breadcrumbs are useful, but they're thin–the http breadcrumb just lets you state that a request happened, and what it's status code was, but not much more (headers, body).